ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -944,30 +1376,31 @@ void CMitsubaLoader::genBSDFtreeTraversal(SContext& ctx, const CElementBSDF* bsd
 		case CElementBSDF::Type::ROUGHPLASTIC:
 			_IRR_FALLTHROUGH;
 		case CElementBSDF::Type::WARD:
-			stack.push({_bsdf,_1stdword,false}); break;
+			stack.push({_bsdf,_1stdword,false,0});
+			break;
 		case CElementBSDF::Type::COATING:
 			_IRR_FALLTHROUGH;
 		case CElementBSDF::Type::ROUGHCOATING:
 			_IRR_FALLTHROUGH;
 		case CElementBSDF::Type::BUMPMAP:
 			_1stdword &= (~(bsdf::INSTR_NORMAL_REG_MASK<<bsdf::INSTR_NORMAL_REG_SHIFT));//zero-out normal reg bitfield
-			assert(firstFreeNormalReg<=bsdf::INSTR_NORMAL_REG_MASK);
+			_IRR_DEBUG_BREAK_IF(firstFreeNormalReg>bsdf::INSTR_NORMAL_REG_MASK);
 			_1stdword |= (firstFreeNormalReg & bsdf::INSTR_NORMAL_REG_MASK) << bsdf::INSTR_NORMAL_REG_SHIFT;//write new val
 			++firstFreeNormalReg;
-			stack.push({_bsdf,_1stdword,false});
+			stack.push({_bsdf,_1stdword,false,0});
 
 			_1stdword = BSDFtype2opcode(_bsdf->coating.bsdf[0]);
 			writeInheritableFlags(_1stdword, _parent1stDword);
-			stack.push({_bsdf->coating.bsdf[0],_1stdword,false});
+			stack.push({_bsdf->coating.bsdf[0],_1stdword,false,0});
 			break;
 		case CElementBSDF::Type::BLEND_BSDF:
-			stack.push({ _bsdf,_1stdword,false });
+			stack.push({ _bsdf,_1stdword,false,0 });
 			_1stdword = BSDFtype2opcode(_bsdf->blendbsdf.bsdf[1]);
 			writeInheritableFlags(_1stdword, _parent1stDword);
-			stack.push({ _bsdf->coating.bsdf[1],_1stdword,false });
+			stack.push({ _bsdf->coating.bsdf[1],_1stdword,false,0 });
 			_1stdword = BSDFtype2opcode(_bsdf->blendbsdf.bsdf[0]);
 			writeInheritableFlags(_1stdword, _parent1stDword);
-			stack.push({ _bsdf->coating.bsdf[0],_1stdword,false });
+			stack.push({ _bsdf->coating.bsdf[0],_1stdword,false,0 });
```

Comment:
why are you using `coating` here everywhere? I'm getting confused

Reply:
sometimes im accessing `bsdf` table through random union member (e.g. coating) because I'd to have lots of ifs just to access it through proper union member even though it's always same array (since it's a union and `bsdf` array is at the same offset in all bsdf structs).
 Well.. this explanation is also kinda blurry XD maybe you'll want to ask on discord

Reply:
can you add another union as an alias, and use that instead so we're not confused to high shit?

Reply:
status?

Reply:
fixed, made this union member
```
			//a not confusing way (extra union member) to access members common for all structs inheriting from MetaBSDF
			MetaBSDF			meta_common;
```